### PR TITLE
Test that loan cannot be repaid with wrong amount/asset/address

### LINKED ISSUE #49: Add more negative tests
We already test that our loan protocol works if the parties follow the script. 

For the dynamic liquidation path, we also test that it cannot be triggered with: a signature from someone other than the oracle; a signature for a price that is above the minimum collateral price; and a signature for a timestamp that is before the contract creation time.

This sort of testing proves that our protocols cannot be exploited with unexpected behaviour. We therefore want to increase the number of these _negative_ tests. For example, we should test that invalid repayment outputs (wrong address, wrong amount, etc.) do not let a borrower take back their collateral.

_Originally posted by @thomaseizinger in https://github.com/comit-network/baru/issues/34#issuecomment-893180604_

### PR DESCRIPTION
**Pull request by luckysori**
I intended to test this using `elements-consensus`, but `elements-miniscript` already gives us an error when trying to satisfy
the miniscript.

We may want to add more tests like this one, so we may not want to close #49 if we merge this.

### CODE REVIEW

File: src/loan/negative_tests.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            TxOut::new_not_last_confidential(
+                &mut rng,
+                SECP256K1,
+                repayment_amount - 5_000,
+                repayment_address.clone(),
+                usdt_asset_id,
+                &inputs,
+            )
+            .unwrap()
+        };
```
Comment (thomaseizinger) [inline src/loan/negative_tests.rs:133]: This seems to be a very repeated piece of code. Can we re-use this somehow?

**Review by thomaseizinger — APPROVED** [review]
Thanks for adding this test!

Can we split this into 3 smaller ones? It should be relatively trivial to extract a common "setup" function so that each test has a visible structure of:

- arrange
- act
- assert

Treating tests as first-class citizens that deserve a nice API might also help you to push for a more modular API design of the library itself, hence reducing test specific setup code again :)

**Comment by thomaseizinger** [comment]
Also, we can't tests live in `tests/`?

**Comment by thomaseizinger** [comment]
> > Also, we can't these tests live in `tests/`?
> 
> We're not testing public API here. I can make `satisfy_loan_repayment` public, but I'm not sure I should.

Right, that makes sense. Whatever falls out of https://github.com/comit-network/baru/discussions/68 should allow us to test this using the public API I think.

**Comment by luckysori** [comment]
Merging after tackling suggestions and adding another test. If you leave any further feedback we can tackle it in a follow-up PR.

**Pull request merged**